broker/broker.go

```diff
@@ -217,35 +229,53 @@ func (b *Broker) Update(ctx context.Context, instanceID string, details brokerap
 		}
 	}
 
-	params := map[string]string{}
-	if userParameters.MaxMemoryPolicy != nil {
-		params["maxmemory-policy"] = *userParameters.MaxMemoryPolicy
+	if userParameters.MaxMemoryPolicy == nil && userParameters.PreferredMaintenanceWindow == "" {
+		return brokerapi.UpdateServiceSpec{}, fmt.Errorf("no parameters provided")
 	}
 
-	if details.PlanID != details.PreviousValues.PlanID {
-		return brokerapi.UpdateServiceSpec{}, fmt.Errorf("changing plans is not currently supported")
-	}
+	var updateRepGrpError error
+	var updateParamsError error
 
-	if details.ServiceID != details.PreviousValues.ServiceID {
-		return brokerapi.UpdateServiceSpec{}, fmt.Errorf("changing plans is not currently supported")
+	if userParameters.PreferredMaintenanceWindow != "" {
+		updateRepGrpError = b.provider.UpdateReplicationGroup(providerCtx, instanceID, providers.UpdateReplicationGroupParameters{
+			PreferredMaintenanceWindow: userParameters.PreferredMaintenanceWindow,
+		})
+		if updateRepGrpError != nil {
+			b.logger.Debug("update-replication-group-success", lager.Data{
+				"instance-id":        instanceID,
+				"details":            details,
+				"accepts-incomplete": asyncAllowed,
+			})
+		}
 	}
 
-	if len(params) == 0 {
-		return brokerapi.UpdateServiceSpec{}, fmt.Errorf("no parameters provided")
+	params := map[string]string{}
+	if userParameters.MaxMemoryPolicy != nil {
+		params["maxmemory-policy"] = *userParameters.MaxMemoryPolicy
 	}
 
-	err := b.provider.Update(providerCtx, instanceID, providers.UpdateParameters{
+	updateParamsError = b.provider.UpdateParams(providerCtx, instanceID, providers.UpdateParamGroupParameters{
 		Parameters: params,
 	})
-	if err != nil {
-		return brokerapi.UpdateServiceSpec{}, err
+	if updateParamsError != nil {
+		b.logger.Debug("update-parameter-group-success", lager.Data{
+			"instance-id":        instanceID,
+			"details":            details,
+			"accepts-incomplete": asyncAllowed,
+		})
+	}
+	var errors []string
+
+	if updateParamsError != nil && userParameters.MaxMemoryPolicy != nil {
+		errors = append(errors, fmt.Sprintf("maxmemory op failed: %v", updateParamsError.Error()))
```

Comment:
Looks a bit like you're reinventing https://pkg.go.dev/github.com/pkg/errors#Wrap here